cmd/frostfs-node/config.go

```diff
@@ -911,60 +911,91 @@ type dCfg struct {
 	}
 }
 
-func (c *cfg) configWatcher(ctx context.Context) {
+func (c *cfg) signalWatcher() {
 	ch := make(chan os.Signal, 1)
-	signal.Notify(ch, syscall.SIGHUP)
+	signal.Notify(ch, syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM)
 
 	for {
 		select {
-		case <-ch:
-			c.log.Info("SIGHUP has been received, rereading configuration...")
-
-			err := c.readConfig(c.appCfg)
-			if err != nil {
-				c.log.Error("configuration reading", zap.Error(err))
-				continue
+		case sig := <-ch:
+			switch sig {
+			case syscall.SIGHUP:
+				c.reloadConfig()
```

Comment:
Actually, it can be problematic handling `SIGHUP` and `SIGTERM` in one thread, because SIGHUP handling can take a long time. However, there are mutexes inside of an engine anyway, so we may postpone this until #49.